logserver/src/main/java/com/yahoo/logserver/handlers/HandlerThread.java

```diff
@@ -201,10 +182,12 @@ public void run() {
                 drainList.add(queue.take());
                 queue.drainTo(drainList);
 
-                for (ItemOrList o : drainList) {
+                for (int i = 0; i < drainList.size(); i++) {
                     // we can get two types of elements here: single log
                     // messages or lists of log messages, so we need to
                     // handle them accordingly.
+                    ItemOrList o = drainList.get(i);
```

Comment:
why do you get() then set() the same thing?

Reply:
Argh, null it should be.